infer/src/atd/sarifbug.atd

```diff
@@ -45,12 +45,17 @@ type codeflow = {
   threadFlows : threadflow list;
 }
 
+type fingerprints = {
+  hashV1 <json name="hash/v1"> : string;
```

Comment:
Any particular reason for it to be named `hashV1` instead of just `hash`?

Reply:
In SARIF, [the fingerprints property](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317654) allows to use different ways of computing issue identification. The idea is that whenever we want to use a different and incompatible hash mechanism here, we can just bump the version to `"hash/v2"`. However, I think it is probably fine to simply use `"hash"` for now and bump to `"hash/v1"` only once we need versioning.

We should probably also add the `key` field as part the fingerprints here. The key is expected to be stable whereas the `hash` could in theory depend on external things (i.e. the version of the compiler?). Adding the `key` here will be useful to compare findings over time, e.g. like:
```
   "fingerprints": {
      "hash" : "4ca6a6a9fe588dc9c25faac035d46f49",
      "key" : "Lists.java|removeInvalidatesNonEmptinessNPE|NULL_DEREFERENCE"
    }
```

Reply:
According to Edoardo, who originally added this property in our own codebase, in order to be compliant with the SARIF spec it would have to be a version hierarchical string. [3.27.16](https://docs.oasis-open.org/sarif/sarif/v2.1.0/csprd01/sarif-v2.1.0-csprd01.html#_Toc10541092), [3.5.4.2)](https://docs.oasis-open.org/sarif/sarif/v2.1.0/csprd01/sarif-v2.1.0-csprd01.html#_Toc10540879).

Unless there is a strong reason to deviate from this, I'd suggest to start out with `hash/v1`. By the way, @geralt-encore, was your question with respect to the json `hash/v1` or just the ATD property name `hashV1`? I suppose for the latter it makes no difference, we could call it `hash` as well if preferred.

Reply:
Updated the key property accordingly:
```json
"fingerprints": {
    "hash/v1": "71dd1e5127da495e4011c5acef571a0e",
    "key": "AnalysisStops.java|fieldReadInCalleeMayCauseFalseNegativeBad|NULLPTR_DEREFERENCE"
}
```